src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource_Async.cs

```diff
@@ -214,48 +221,52 @@ await foreach (var set in allSets)
 
                 yield break;
 
-                Task<ImmutableArray<UnifiedSuggestedActionSet>> GetCodeFixesAsync()
+                async Task<ImmutableArray<UnifiedSuggestedActionSet>> GetCodeFixesAsync()
                 {
+                    using var _ = TelemetryLogging.LogBlockTimeAggregated(FunctionId.SuggestedAction_Summary, telemetryContext + ".GetCodeFixesAsync");
+
                     if (owner._codeFixService == null ||
                         !supportsFeatureService.SupportsCodeFixes(target.SubjectBuffer) ||
                         !requestedActionCategories.Contains(PredefinedSuggestedActionCategoryNames.CodeFix))
                     {
-                        return SpecializedTasks.EmptyImmutableArray<UnifiedSuggestedActionSet>();
+                        return ImmutableArray<UnifiedSuggestedActionSet>.Empty;
                     }
 
-                    return UnifiedSuggestedActionsSource.GetFilterAndOrderCodeFixesAsync(
+                    return await UnifiedSuggestedActionsSource.GetFilterAndOrderCodeFixesAsync(
                         workspace, owner._codeFixService, document, range.Span.ToTextSpan(),
-                        priorityProvider, options, addOperationScope, cancellationToken).AsTask();
+                        priorityProvider, options, addOperationScope, cancellationToken).ConfigureAwait(false);
                 }
 
-                Task<ImmutableArray<UnifiedSuggestedActionSet>> GetRefactoringsAsync()
+                async Task<ImmutableArray<UnifiedSuggestedActionSet>> GetRefactoringsAsync()
                 {
+                    using var _ = TelemetryLogging.LogBlockTimeAggregated(FunctionId.SuggestedAction_Summary, telemetryContext + ".GetRefactoringsAsync");
```

Comment:
Can we also use `nameof(GetRefactoringsAsync)` here like you did at other places?